src/Key.php

```diff
@@ -5,18 +5,132 @@
 class Key
 {
     /** @var string */
-    public $id;
+    private $id;
 
     /** @var string */
-    public $secret;
+    private $secret;
+
+    /** @var string */
+    private $certificate;
+
+    /** @var string */
+    private $privateKey;
+
+    /** @var string */
+    private $type;
 
     /**
      * @param string $id
      * @param string $secret
      */
-    public function __construct($id, $secret)
+    public function __construct($id, $item)
     {
         $this->id = $id;
-        $this->secret = $secret;
+        $certificate = $this->getX509Certificate($item);
+        $privateKey = $this->getRSAPrivateKey($item);
+        if (($certificate || $privateKey)) {
```

Comment:
this if statement seems unnecessary as your nested `if` check the same thing. Try to avoid nesting control statements

Reply:
HMAC only takes a single key for signing and verifying, RSA/EC require two distinct keys, so this approach allows the existing Key class and KeyStore interface to be used for both since we don't have an explicit method for supplying one or the other. Alternative is to break Key into different subtypes (e.g. HMAC & PKI or Shared, Private & Public), but this flow is tolerant of whatever the user throws into the input array - this could be a private key, a certificate, or both.